### PR TITLE
Dashboards: fix autoscaling panels

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does

Autoscaling panels queries are wrong after #9412, due to a couple of main issues:

1. One query is syntactically wrong
2. Aggregations are wrong, we need to aggregate on different labels or aggregate after relabelling (I preferred to aggregate after relabelling to fix it)

Preview of fixed panels:
Screenshot 2024-09-27 at 15 37 34

Screenshot 2024-09-27 at 15 37 29

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: operations/mimir-mixin/dashboards/dashboard-utils.libsonnet [hunk 1]
```diff
@@ -701,20 +701,25 @@ local utils = import 'mixin-utils/utils.libsonnet';
...
           sum by (scaler) (
-            label_replace(
-              # Using `max by ()` so that series churn doesn't break the promQL join
-              max by (%(aggregation_labels)s, scaledObject, metric, scaler) keda_scaler_metrics_value{%(cluster_label)s=~"$cluster", exported_namespace=~"$namespace", scaler=~"%(scaler_selector)s"},
-              "namespace", "$1", "exported_namespace", "(.*)"
-            )
-            /
-            on(%(aggregation_labels)s, scaledObject, metric) group_left label_replace(
+            # Using `max by ()` so that series churn doesn't break the promQL join
+            max by (%(aggregation_labels)s, scaledObject, metric, scaler) (
```
Comment (dimitarvdimitrov) [inline operations/mimir-mixin/dashboards/dashboard-utils.libsonnet:705]: should this also include `namespace`?

File: operations/mimir-mixin/dashboards/dashboard-utils.libsonnet [hunk 2]
```diff
@@ -746,26 +751,34 @@ local utils = import 'mixin-utils/utils.libsonnet';
...
           sum by (scaler) (
-            label_replace(
-              # Using `max by ()` so that series churn doesn't break the promQL join
-              max by (namespace, scaler, %(aggregation_labels)s, scaledObject, metric) (keda_scaler_metrics_value{%(cluster_label)s=~"$cluster", exported_namespace=~"$namespace", scaler=~"%(scaler_selector)s"}),
-              "namespace", "$1", "exported_namespace", "(.*)"
-            )
-            /
-            on(%(aggregation_labels)s, scaledObject, metric) group_left label_replace(
+            # Using `max by ()` so that series churn doesn't break the promQL join
+            max by (%(aggregation_labels)s, scaledObject, metric, scaler) (
```
Comment (dimitarvdimitrov) [inline operations/mimir-mixin/dashboards/dashboard-utils.libsonnet:755]: also namespace here?

File: operations/mimir-mixin/dashboards/dashboard-utils.libsonnet (hunk 1 above)
Comment (pracucci) [reply operations/mimir-mixin/dashboards/dashboard-utils.libsonnet:705]: namespace is already part of the aggregation_labels

File: operations/mimir-mixin/dashboards/dashboard-utils.libsonnet (hunk 2 above)
Comment (pracucci) [reply operations/mimir-mixin/dashboards/dashboard-utils.libsonnet:755]: namespace is already part of the aggregation_labels

**Review by dimitarvdimitrov — APPROVED** [review]

**Pull request merged**